filters/filter.h

```diff
@@ -398,6 +398,7 @@ struct mp_stream_info {
     void *priv; // for use by whoever implements the callbacks
 
     double (*get_display_fps)(struct mp_stream_info *i);
+    void   (*get_display_res)(struct mp_stream_info *i, void *res);
```

Comment:
why not `int64_t *`?

Reply:
I don't think this can be in64_t * without allocating memory? That's why I went with void pointers.

Reply:
Why would a int64_t pointer require an allocation when a void pointer can do it without? The void pointer still has to point somewhere, and at the place where it's called there is a int64_t array. that ends up getting pointed to. I don't see how the type changes anything here regarding memory.

Reply:
Maybe I was just returning the wrong address. Let me try again.

Reply:
Yeah, not sure how to do it like this. Maybe I'm just bad at C but the in64_t array is always local and I'm not sure how you would go about returning it. Plus there's also an additional conversion step you would have to do because VOCTRL_GET_DISPLAY_RES requires int[2] not int64_t[2].

Reply:
Then the array we pass in also needs to be int[2] and then converted into int64_t[2]. Making the pointer void doesn't change the location of the values. Alternatively we could also change VOCTRL_GET_DISPLAY_RES to int64_t[2], not sure if that's viable.

Reply:
Yeah it was returning garbage values. Should be fixed now.

>VOCTRL_GET_DISPLAY_RES to int64_t[2]

Not worth it imo.

Reply:
Agree

Reply:
ah sorry I misread, this can be `int *` however

nothing to do with allocations, it's just the type that's actually being used.

Reply:
I just needed to make it static.